developer_manual/basics/storage/database.rst

```diff
@@ -321,6 +321,37 @@ Slugs are used to identify resources in the URL by a string rather than integer
     $author->setName('Some*thing');
     $author->slugify('name');  // Some-thing
 
+Table management tips
+-------------------------
+
+It makes sense to apply some general tips from the beginning, so you don't have to migrate your data and schema later on.
+
+1. Don't use table name longer than 23 characters. As Oracle is limited to 30 chars and we need 3 more for `oc_` at the beginning and 5 for the primary key suffix `_pkey`.
+
+2. Add an auto-incremented `id` column. This will ease the use of `QBMapper` + `Entity` approach:
+    - https://github.com/nextcloud/server/blob/master/lib/public/AppFramework/Db/QBMapper.php
+    - https://github.com/nextcloud/server/blob/master/lib/public/AppFramework/Db/Entity.php
+
+.. code-block:: php
+
+    $table->addColumn('id', Types::BIGINT, [
+        'autoincrement' => true,
+        'notnull' => true,
+        'length' => 20,
+        'unsigned' => true,
+    ]);
+
+3. Set a primary key to prevent errors in clustered setups. You can use the `id` field for that.
+
+.. code-block:: php
+
+    $table->setPrimaryKey(['id']);
+
+4. Manually set the name of your indexes. It will help you to manipulate them if needed in the future. Note that the names of the index are "global" database wide in some DBs. So having generic names can create conflicts.
```

Comment:
```suggestion
4. Manually set the name of your indexes. It will help you to manipulate them if needed in the future. Note that the names of the index are "global" database wide in some database platforms. So having generic names can create conflicts.
```